torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -884,59 +896,19 @@ std::vector<int64_t> TensorDomain::mapDomainCtoP(
 std::unordered_map<IterDomain*, IterDomain*> TensorDomain::mapRootCtoP(
     const TensorDomain* consumer,
     const TensorDomain* producer,
-    bool constrain,
     const std::unordered_set<IterDomain*>& consumer_root_dims_to_map) {
-  auto consumer_root = consumer->rootDomain();
-  auto producer_root = producer->hasRFactor() ? producer->rfactorDomain()
-                                              : producer->rootDomain();
-
-  auto c_to_p = mapDomainCtoP(consumer_root, producer_root);
-
   std::unordered_map<IterDomain*, IterDomain*> root_id_map;
-
-  for (int64_t itc = 0; itc < (int64_t)c_to_p.size(); itc++) {
-    int64_t itp = c_to_p[itc];
-    if (itp == -1)
-      continue;
-
-    if (!constrain ||
-        (constrain &&
-         consumer_root_dims_to_map.find(consumer_root[itc]) !=
-             consumer_root_dims_to_map.end())) {
-      root_id_map[consumer_root[itc]] = producer_root[itp];
+  for (const auto& kv : mapRootPandC(producer, consumer)) {
+    auto producer_axis = kv.first;
+    auto consumer_axis = kv.second;
+    if (consumer_root_dims_to_map.find(consumer_axis) !=
+        consumer_root_dims_to_map.end()) {
+      root_id_map[consumer_axis] = producer_axis;
     }
   }
   return root_id_map;
 }
 
-// return mapping of consumer_domain[i] = producer_domain[result_vector[i]]
-// assuming there exists a direct consumer-producer mapping. If axis exists in
-// consumer (broadcast) but not in producer, mapping will be result_vector[i] =
-// -1.
-std::vector<int64_t> TensorDomain::mapDomainPtoC(
-    const std::vector<IterDomain*>& producer,
-    const std::vector<IterDomain*>& consumer) {
-  std::vector<int64_t> producer_to_consumer(producer.size(), -1);
-
-  size_t itc = 0, itp = 0;
-  while (itc < consumer.size() && itp < producer.size()) {
-    if (consumer[itc]->isBroadcast() && !producer[itp]->isBroadcast()) {
-      itc++;
-      continue;
-    }
-    if (producer[itp]->isReduction()) {
-      itp++;
-      continue;
-    }
-
-    producer_to_consumer[itp] = itc;
-    itc++;
-    itp++;
-  }
-
-  return producer_to_consumer;
-}
-
 // Create a map from producer root IterDomains -> consumer root IterDomains.
 // Constrain will restrict which producer root IterDomains we map to the
```

Comment:
Update this comment please.

Reply:
I just deleted it as it's redundant to have the same comment both in the header and source files.